ipv6/prefix.go

```diff
@@ -0,0 +1,208 @@
+package ipv6
+
+import (
+	"fmt"
+	"net"
+)
+
+// Prefix represents an IP prefix which is formally an Address plus a Mask. It
+// is stored in a more space-efficient way by storing the number of 1s in the
+// Mask as a length.
+// Note that any bits in the address can be 0 or 1 regardless if they in the
+// first `length` bits or not. This allows storing an IP address in CIDR
+// notation with both the network and host parts of the address.
+// The zero value of a Prefix is "0.0.0.0/0"
+type Prefix struct {
+	addr   Address
+	length uint32
```

Comment:
I wonder why I went with uint32 here on the IPv6 side. I honestly don't remember. Maybe it is because on a 32-bit processor, this would allow the address + length to align within 8 bytes.

Now, I'm curious to know if going with a uint16 would allow us to realign the [trieNode struct](https://github.com/go-addrs/addrs/blob/v0.30.0/ipv4/node.go#L9-L14) and make it smaller. Is that something you'd want to play with? The thing that I don't know is whether we can use the white space in the Prefix struct to pack some of the smaller fields from the trieNode struct. I dunno. I guess we'll have to write a test to know. Nothing for this PR, just pontificating here...

I don't remember all of the alignment rules but I do remember playing with it quite a bit in a previous incarnation of this code. I went as far as to [write a unit test](https://github.com/ecbaldwin/trie/blob/e195ac48b6f6659dd35f4383aad228abb14d4939/internal/trienode32_test.go#L19-L54) that would assert the size of a struct on both 32 bit and 64 bit architectures.

Reply:
Yeah I can mess around with the size of it when I start going through node.go